src/framework/mlt_chain.c

```diff
@@ -393,6 +394,56 @@ void mlt_chain_close( mlt_chain self )
 	}
 }
 
+/** Attach normalizer links.
+ *
+ * \public \memberof mlt_chain_s
+ * \param self a chain
+ */
+
+extern void mlt_chain_attach_normalizers( mlt_chain self )
+{
+	if ( !self ) return;
+
+	if ( mlt_chain_link_count( self ) && mlt_properties_get_int( MLT_LINK_PROPERTIES( mlt_chain_link( self, 0 ) ), "_loader" ) )
+	{
+		// Chain already has normalizers
+		return;
+	}
+
+	static mlt_properties normalisers = NULL;
+
+	mlt_tokeniser tokenizer = mlt_tokeniser_init( );
+
+	// We only need to load the normalising properties once
+	if ( normalisers == NULL )
+	{
+		char temp[ 1024 ];
+		sprintf( temp, "%s/chain_normalizers.ini", mlt_environment( "MLT_DATA" ) );
```

Comment:
Please use `PATH_MAX` instead of hardcoded 1024, and `snprintf()`. I realize that is not being used everywhere, in particular the loader producer, and I will make a separate change to address that.

Reply:
done